### PR TITLE
Add group pseudonymization support (fixes #14117)

### LINKED ISSUE #14117: Pseudonymize groups
Our pseudonymization mechanism in [`org.jabref.logic.pseudonymization.Pseudonymization`](https://github.com/JabRef/jabref/blob/main/jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java) is not exhaustive - for instance, it misses the pseudonymization of [groups](https://docs.jabref.org/finding-sorting-and-cleaning-entries/groups).
The goal is to add that.

**Additional context**
- JabRef groups: https://docs.jabref.org/finding-sorting-and-cleaning-entries/groups
- The information about groups is contained in the `.bib` file itself, at the bottom, with `@Comment`.
- One can explore the above using [Chocolate.bib](https://github.com/JabRef/jabref/blob/main/jablib/src/main/resources/Chocolate.bib) and co-relate with what is happening in the GUI (groups section, and also by adding new groups).
- The entry point to use pseudonymization is currently under JabKit (CLI), with hints to use in [this](https://github.com/JabRef/jabref/pull/13158) PR that added it.
- Use JabRef's classes to interact with groups (this also updates `@Comment` content)
- Remember that `groups` field has **multiple** groups - and the names are the links into the groups tree.

> [!NOTE]
> A **screen recording** in the description of the PR of this functionality working is **mandatory** for marking completion. That is, the output bib file and the pseudonymized groups should be visible in the recording.

### PR DESCRIPTION
**Pull request by f0restron07**
## Description
added group pseudonymization before this only entries were pseudonymized now groups get pseudonymized too for eg group-1,group-2,...etc

## Changes
- added pseudonymizeGroupTree() method which uses deepCopy() and nameProperty().set() so in order the  original library stays unchanged and group types (SearchGroup, KeywordGroup etc) are been preserved
- extracted groups field handling into separate method named as pseudonymizeGroupsField() instead of manual splitting
- groups in entries which doesn't exist in the group tree also get pseudonymized  with a counter
- added changelog entry

## Testing
- All test pass
- Added shouldPseudonymizeGroupTree() test
- Added shouldPseudonymizeUnmappedGroups() test for edge case where group in entry doesnt exist in tree

Before (original groups):

image

After (pseudonymized groups):

image

Fixes #14117

### AI Disclosure
I used Chatgpt AI to assist with understanding the codebase and debugging.

### Checklist
- [x] I own the copyright of the code submitted and I license it under the MIT license
- [x] I manually tested my changes in running JabRef (always required)
- [x] I added JUnit tests for changes (if applicable)
- [x] I added screenshots in the PR description (if change is visible to the user)
- [ ] I added a screenshot in the PR description showing a library with a single entry with me as author and as title the issue number
- [x] I described the change in CHANGELOG.md in a way that can be understood by the average user (if change is visible to the user)
- [ ] I checked the user documentation for up to dateness and submitted a pull request to our user documentation repository

### CODE REVIEW

**Comment by f0restron07** [comment]
https://github.com/user-attachments/assets/790fe44f-1f54-4d1d-8d15-0f12649d0d08

**Review by koppor — CHANGES_REQUESTED** [review]
- Tests missing
- file field removed
- int[] is a hack. Should be made differentlt


"Steps to test" removed. Indicator for too much AI usage. Therefore only brief review.

**Comment by f0restron07** [comment]
Thanks for the review.. I'll address each of the pt:
1. Add unit tests
2. Restore file fields
3. Replace int[] with AtomicInteger
Will push the fixes soon

**Comment by f0restron07** [comment]
Hey @koppor 
I have done the changes:
1.  Replaced int[] with `AtomicInteger`
2.  Added a unit test shouldPseudonymizeGroupTree() for group pseudonymization
Regarding the file field: The pseudonymizeLibrary test was already failing before my code changes because the output does not include file fields. I removed the file fields from the expected file

Should I change this back and leave this as a separate issue?

**Comment by koppor** [comment]
> 3. Replace int[] with AtomicInteger

Is strange. Isn't it possible with normal recursion?

In tests we use "withField", not setField

"pt" in comment text strange

**Comment by subhramit** [comment]
> * All existing tests pass
> * Groups are correctly pseudonymized in output

No screenshot or video recording is provided in the description. Please add them.

**Comment by f0restron07** [comment]
@koppor 
I have implemented all the suggestions:
- Normal recursion instead of using AtomicInteger
- Used withField in tests
- Fixed the comment
- Screenshots added

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 1]
```diff
@@ -37,6 +41,13 @@ public Result pseudonymizeLibrary(BibDatabaseContext bibDatabaseContext) {
         BibDatabaseContext result = new BibDatabaseContext(bibDatabase);
         result.setMode(bibDatabaseContext.getMode());
 
+        Optional<GroupTreeNode> groups = bibDatabaseContext.getMetaData().getGroups();
+        if (groups.isPresent()) {
+            int[] counter = {1};
+            GroupTreeNode newGroups = pseudonymizeGroupTree(groups.get(), valueMapping, counter);
+            result.getMetaData().setGroups(newGroups);
+        }
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java]: The bot is right here

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 2]
```diff
@@ -59,4 +70,18 @@ private static List<BibEntry> pseudonymizeEntries(BibDatabaseContext bibDatabase
...
         return newEntries;
     }
+
+    private static GroupTreeNode pseudonymizeGroupTree(GroupTreeNode node, Map<String, String> valueMapping, int[] counter) {
+        AbstractGroup originalGroup = node.getGroup();
+        String originalName = originalGroup.getName();
+        String pseudoName = "group-" + counter[0];
+        counter[0]++;
+        valueMapping.put(pseudoName, originalName);
+        AbstractGroup newGroup = new ExplicitGroup(pseudoName, originalGroup.getHierarchicalContext(), ',');
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java]: The bot is right here, this replaces the group types

**Comment by f0restron07** [comment]
@Siedlerchr  I've applied these two fixes:-
1. Preserved the AllEntriesGroup for the root
2. Aligned the entry groups field with the group tree names (Used → group-8)
This means we need to update the test file should i continue with the update of `Chocolate-pseudnomyized.bib` to match the expected behavior?

**Comment by Siedlerchr** [comment]
Yes sure

Omkar Ghotne ***@***.***> schrieb am So., 8. März 2026, 08:24:

**Comment by f0restron07** [comment]
@Siedlerchr @koppor I've addressed the query
1. The root AllEntriesGroup was preserved.
2. Aligning the group tree names with the entry groups field
3. The expected test file has been updated.

**Comment by subhramit** [comment]
> @Siedlerchr @koppor I've addressed the query
> 
> 1. The root AllEntriesGroup was preserved.
> 2. Aligning the group tree names with the entry groups field
> 3. The expected test file has been updated.

If you use AI for communication one more time, this PR will be closed and you will be blacklisted from taking further assignments.
Please understand your changes and communicate in your own way based on your own understanding.

**Comment by f0restron07** [comment]
@subhramit sorry for the issue  i'm new to open source so in order to sound a bit good i took the help of Ai for communication extremely sorry for the mishap wont happen again . so the changes i have made in this pr are as follows-:

- so i added a pseudonymizeGroupTree method where in it turns the groups like"used,read,skimmed" to "group-1,group2,group3 etc" 
- and also made an groupNameMapping hashmap  in order to track which group has become what so when processing the entries if group="used" it looks up in the map and replaces with "group-8" so in this manner group tree and entry have the same name
- @Siedlerchr pointed out that the old code was converting all groups to explicit group but the root group is AllEntriesGroup and that should stay that away so i added a check where if group is AllEntriesGroup keep it as AllEntriesGroup otherwise make explicitgroup
- and also since the code output changed due to point 3 the expected test file also needed to  changed so i seeked permission from then changed chocolate-pseudnomyized.bib to match new output

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 3]
```diff
@@ -59,4 +94,34 @@ private static List<BibEntry> pseudonymizeEntries(BibDatabaseContext bibDatabase
...
+        String pseudoName = "group-" + counter;
+        counter++;
+        valueMapping.put(pseudoName, originalName);
+        groupNameMapping.put(originalName, pseudoName);
+
+        AbstractGroup newGroup;
+        if (originalGroup instanceof AllEntriesGroup) {
+            newGroup = new AllEntriesGroup(pseudoName);
+        } else {
+            newGroup = new ExplicitGroup(pseudoName, originalGroup.getHierarchicalContext(), ',');
```
Comment (Siedlerchr) [inline jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java]: I think Qodo bot also told this, but JabRef supports different group types. You are just replacign them with Explicit Groups
Also I wonder why you need to re-create the groups. Why not just change the name of the group via 
nameProperty?

Comment (f0restron07) [reply jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java]: sry i did use explicit groups because i didn't knew about the nameProperty thing that's why i was creating new group object every time so i re-checked using nameProperty but that thing was changing the orignal library which it should not. so i used DeepCopy() to make clone of the group then renamed the clone using nameProperty.set() this way the orignal library stays as it is and grouptypes also preserved in their orignal form .to match the output i also updated the choclate-pseudnomyized.bib

**Comment by f0restron07** [comment]
@Siedlerchr @koppor just following up i've actually addressed the usage of explicit groups issue by resolving it using deepcopy() in order to clone the group and nameProperty.set to rename it so in this way the orignal library stays unchanged and group types are intact. all the test have been passed could u please re-review it

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 4]
```diff
@@ -26,29 +30,58 @@ public Result pseudonymizeLibrary(BibDatabaseContext bibDatabaseContext) {
...
         List<BibEntry> newEntries = new ArrayList<>(entries.size());
 
         for (BibEntry entry : entries) {
             BibEntry newEntry = new BibEntry(entry.getType());
             newEntries.add(newEntry);
             for (Field field : entry.getFields()) {
+                if (field.equals(StandardField.GROUPS)) {
+                    String fieldContent = entry.getField(field).orElse("");
+                    if (!fieldContent.isBlank()) {
+                        String[] groupNames = fieldContent.split(",");
```
Comment (koppor) [inline jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java]: I think, its the wrong place to parse things.

Use org.jabref.model.entry.KeywordList#parse.

You can also add a link to `org.jabref.model.entry.field.StandardField#GROUPS` so that others find this, too.

You could also try to update `docs/code-howtos/groups.md`.

File: jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java [hunk 5]
```diff
@@ -56,43 +60,51 @@ void setUp() {
...
     @Test
     void pseudonymizeTwoEntries() {
-        BibEntry first = new BibEntry("first")
-                .withField(StandardField.AUTHOR, "Author One")
-                .withField(StandardField.PAGES, "some pages");
-        BibEntry second = new BibEntry("second")
-                .withField(StandardField.AUTHOR, "Author Two")
-                .withField(StandardField.PAGES, "some pages");
+        BibEntry first = new BibEntry("first").withField(StandardField.AUTHOR, "Author One").withField(StandardField.PAGES, "some pages");
+        BibEntry second = new BibEntry("second").withField(StandardField.AUTHOR, "Author Two").withField(StandardField.PAGES, "some pages");
```
Comment (koppor) [inline jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java:81]: No code reformatting in PRs please - the new code style is not as readble as the old style, because at the old style, each key is on a separate line.

You are invited to update the code, because "first" and "second" are entry types, but it should be citation keys.

File: jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java [hunk 6]
```diff
@@ -56,43 +60,51 @@ void setUp() {
...
         Pseudonymization.Result result = pseudonymization.pseudonymizeLibrary(databaseContext);
 
-        BibEntry firstPseudo = new BibEntry("citationkey-1")
-                .withField(StandardField.AUTHOR, "author-1")
-                .withField(StandardField.PAGES, "pages-1");
-        BibEntry secondPseudo = new BibEntry("citationkey-2")
-                .withField(StandardField.AUTHOR, "author-2")
-                .withField(StandardField.PAGES, "pages-1");
+        BibEntry firstPseudo = new BibEntry("citationkey-1").withField(StandardField.AUTHOR, "author-1").withField(StandardField.PAGES, "pages-1");
+        BibEntry secondPseudo = new BibEntry("citationkey-2").withField(StandardField.AUTHOR, "author-2").withField(StandardField.PAGES, "pages-1");
```
Comment (koppor) [inline jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java]: See above

File: jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java [hunk 7]
```diff
@@ -56,43 +60,51 @@ void setUp() {
...
         BibDatabaseContext bibDatabaseContextExpected = new BibDatabaseContext(new BibDatabase(List.of(firstPseudo, secondPseudo)));
         bibDatabaseContextExpected.setMode(BibDatabaseMode.BIBLATEX);
-        Pseudonymization.Result expected = new Pseudonymization.Result(
-                bibDatabaseContextExpected,
-                Map.of("author-1", "Author One", "author-2", "Author Two", "pages-1", "some pages", "citationkey-1", "first", "citationkey-2", "second"));
+        Pseudonymization.Result expected = new Pseudonymization.Result(bibDatabaseContextExpected, Map.of("author-1", "Author One", "author-2", "Author Two", "pages-1", "some pages", "citationkey-1", "first", "citationkey-2", "second"));
```
Comment (koppor) [inline jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java]: Please keep formatting

File: jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java [hunk 8]
```diff
@@ -56,43 +60,51 @@ void setUp() {
...
 
         assertEquals(expected, result);
     }
 
+    @Test
+    void shouldPseudonymizeGroupTree() {
+        // create simple entry so database isn't empty
+        BibEntry entry = new BibEntry("test")
```
Comment (koppor) [inline jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java]: USe `withCitationKey"` instead of entry type

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 9]
```diff
@@ -26,29 +30,58 @@ public Result pseudonymizeLibrary(BibDatabaseContext bibDatabaseContext) {
...
+                                                      Map<String, String> groupNameMapping) {
         List<BibEntry> entries = bibDatabaseContext.getEntries();
         List<BibEntry> newEntries = new ArrayList<>(entries.size());
 
         for (BibEntry entry : entries) {
             BibEntry newEntry = new BibEntry(entry.getType());
             newEntries.add(newEntry);
             for (Field field : entry.getFields()) {
+                if (field.equals(StandardField.GROUPS)) {
+                    String fieldContent = entry.getField(field).orElse("");
```
Comment (koppor) [inline jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java:80]: Extract to a sepeate method.

**Review by koppor — COMMENTED** [review]
Missing CHANGELOG.md entry

Missing re-use of existing code

Unnecessary code refomatting.

Missing test for "mal-formed" .bib file. - What happens if there is a group `not-in-groups-tree` in `groups` field, which is not in the groups tree? (It should also be pseudonomized!)

File: CHANGELOG.md [hunk 10]
```diff
@@ -29,6 +30,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We replaced the unlinked files dialog with a wizard-based interface for searching and importing files. [#12709](https://github.com/JabRef/jabref/issues/12709)
 - We replaced the various notifications for file changes, tasks and popup toasts with a new info center. [#14762](https://github.com/JabRef/jabref/issues/14762)
 - We upgraded to Lucene 10.4 for the fulltext search.
+
```
Comment (subhramit) [inline CHANGELOG.md]: Why this newline?

**Comment by subhramit** [comment]
> @koppor i have fixed all the things you pointed out for KeywordList#parse issue instead of using split(",") im using the inbuilt KeywordList.parse() method, then created a separate pseudonymizeGroupsField() method to handle the groups field which makes it cleaner, for the test formatting part i put each withField() back on it own line like the original, also fixed the citation key issue by changing new BibEntry("first") which was treating first as an entry type .to new BibEntry().withCitationKey("first") which correctly sets it as the citation key, also added the shouldPseudonymizeUnmappedGroups() test which handles when an entry has a group in its groups field that doesnt exist in the group tree, and lastly added a [changelog.md](http://changelog.md/) entry in the ###added part all the test cases have been passing could u review it ?

Did you see comment https://github.com/JabRef/jabref/pull/15258#issuecomment-4112750249?

**Comment by subhramit** [comment]
> * [x]  I own the copyright of the code submitted and I license it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
> * [x]  Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
> * [x]  Tests created for changes (if applicable)
> * [x]  Screenshots added in PR description (for UI changes)
> * [x]  [Checked developer documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I created an issue at https://github.com/JabRef/jabref/issues.
> * [ ]  [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at https://github.com/JabRef/jabref/issues.
> * [x]  No AI-generated code used OR the AI usage is clearly documented in this PR description.

Why is the wording of the checklist modified? Our CI is failing because of it

**Comment by f0restron07** [comment]
hi @subhramit sry i had added a new line by mistake while editing the document and also i have fixed the Cl issue

**Comment by subhramit** [comment]
> a new line by mistake

Just FYI: The wording of all the sentences were changed, not just a newline. Our CI uses regex matching to look for specific terms.

**Comment by f0restron07** [comment]
@subhramit ya ik in starting of this pr i had used ai in order to communicate and understand the issue due to which the cl had got changed but after your initial warning i completely stopped using it and started writing on my own due to which a part of it was gone unnoticed by me sry for the mistake wont happen again

**Comment by subhramit** [comment]
> @subhramit ya ik in starting of this pr i had used ai in order to communicate and understand the issue due to which the cl had got changed but after your initial warning i completely stopped using it and started writing on my own due to which a part of it was gone unnoticed by me sry for the mistake wont happen again
> 

That's fine. I just wanted to let you know how the tests work.

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java (hunk 9 above)
Comment (koppor) [reply jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java:80]: TODO

Comment (f0restron07) [reply jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java:80]: @koppor i had already fixed the issue in my commit 28a814a .i extracted the group logic in pseudonymizeGroupsField() method down the file so lmk if any changes are required

File: jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java (hunk 7 above)
Comment (calixtus) [reply jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java]: Did some manual reformatting in PseudonymizationTest, which does imho make some things better readable.
Map.of is a badly implementend in JDK - follows pattern:
```
Map.of(key1, value1, key2, value2, key3, value3...)
```

My solution is to group key and value and separate by newlines

**Comment by InAnYan** [comment]
Hi @f0restron07, what's the status of the pr? I quickly looked at the code at seems to be okay

**Comment by f0restron07** [comment]
> Hi @f0restron07, what's the status of the pr? I quickly looked at the code at seems to be okay

Will resolve it in 2 days had some personal issues going on sry for the delay

**Comment by f0restron07** [comment]
@koppor i have addressed all the issues u listed prior so just following up for the re-review and wanted to know if anything needed to change

File: CHANGELOG.md [hunk 11]
```diff
@@ -36,6 +37,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We changed the groups filter field to use a filter icon. [#15402](https://github.com/JabRef/jabref/issues/15402)
 - We improved the MultiMergeEntries dialog to automatically select the most plausible field value when merging entries (e.g. a more specific date). [#15027](https://github.com/JabRef/jabref/issues/15027)
 - We removed the restart prompt when accepting Mr. DLib privacy settings or hiding the Related articles tab in the entry editor. [#15195](https://github.com/JabRef/jabref/issues/15195)
+- We added pseudonymization of group names in the groups field of entries. [#14117](https://github.com/JabRef/jabref/issues/14117)
```
Comment (subhramit) [inline CHANGELOG.md:40]: Why two changelog entries?

Comment (f0restron07) [reply CHANGELOG.md]: typo error sry my bad had got confused b/w added and changed in the start of editing the changelog file had pasted in both added and changed sections but after rectifying it i had removed it .but while undoing some changes i guess it had been restored sry for the blunder will correct it

File: jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java [hunk 12]
```diff
@@ -66,10 +70,12 @@ void setUp() {
 
     @Test
     void pseudonymizeTwoEntries() {
-        BibEntry first = new BibEntry("first")
+        BibEntry first = new BibEntry()
+                .withCitationKey("first")
```
Comment (subhramit) [inline jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java:74]: I'd like more realistic values in tests but the existing ones have similar values so not a blocker

**Review by subhramit — DISMISSED** [review]

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 13]
```diff
@@ -59,4 +87,50 @@ private static List<BibEntry> pseudonymizeEntries(BibDatabaseContext bibDatabase
...
+            String originalName = keyword.toString();
+            String pseudoName = groupNameMapping.get(originalName);
+            if (pseudoName == null) {
+                pseudoName = "group-" + (groupNameMapping.size() + 1);
+                groupNameMapping.put(originalName, pseudoName);
+                valueMapping.put(pseudoName, originalName);
+            }
+            pseudonymizedNames.add(pseudoName);
+        }
+        return String.join(", ", pseudonymizedNames);
```
Comment (koppor) [inline jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java]: REad about groups seperator and update aaccoringly.

**Review by koppor — DISMISSED** [review]

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java (hunk 13 above)
Comment (koppor) [reply jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java]: dont have keywords lists some serialization feture?

**Comment by subhramit** [comment]
please don't rebase/force-push. 
Make any changes you want in subsequent commits.

**Comment by f0restron07** [comment]
@subhramit @koppor sorry for the force push stuff earlier didnt intend to do it but i needed to recover from corrupted merge state had pulled a bad rebase of unrelatd commit from main .
also i have fixed the keyword separator threaded through constructor applied to multi keyword field and keyword serialization using keywordlist#parse+keywordlist.tostring to handle separator instead of hardcoded ","

File: jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java [hunk 14]
```diff
@@ -66,39 +70,86 @@ void setUp() {
...
+        BibDatabase db = new BibDatabase();
+        db.insertEntry(entry);
+        BibDatabaseContext context = new BibDatabaseContext(db);
+        GroupTreeNode root = GroupTreeNode.fromGroup(new AllEntriesGroup("All entries"));
+        GroupTreeNode child = GroupTreeNode.fromGroup(
+                new ExplicitGroup("Research", GroupHierarchyType.INDEPENDENT, ',')
+        );
+        root.addChild(child);
+        context.getMetaData().setGroups(root);
+        Pseudonymization.Result res =
```
Comment (InAnYan) [inline jablib/src/test/java/org/jabref/logic/pseudonymization/PseudonymizationTest.java:123]: A nitpick, but I would try to keep the testing code in 3 blocks: "setup", "action", "check" and separate them by new lines

File: jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 15]
```diff
@@ -59,4 +93,50 @@ private static List<BibEntry> pseudonymizeEntries(BibDatabaseContext bibDatabase
...
+    private String pseudonymizeGroupsField(String fieldContent,
+                                           Map<String, String> groupNameMapping,
+                                           Map<String, String> valueMapping) {
+        KeywordList keywords = KeywordList.parse(fieldContent, keywordSeparator);
+        KeywordList pseudonymizedKeywords = new KeywordList();
+        for (Keyword keyword : keywords) {
+            String originalName = keyword.toString();
+            String pseudoName = groupNameMapping.get(originalName);
+            if (pseudoName == null) {
+                pseudoName = "group-" + (groupNameMapping.size() + 1);
```
Comment (InAnYan) [inline jablib/src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java]: `"group-"` should become a constant `GROUP_PREFIX`

**Comment by subhramit** [comment]
I get
image

When opening a pseudonymized Chocolate.bib.

Also follow-up issue to be created: let a user choose the name of the pseudonymized bibfile (out of scope for this PR)

**Comment by subhramit** [comment]
> I get image
> 
> When opening a pseudonymized Chocolate.bib.

Reason: the line
```
@Comment{jabref-meta: groups-search-syntax-version:6.0-alpha_1}
```
is not preserved by the pseudonymization

**Comment by subhramit** [comment]
> Reason: the line
> 
> ```
> @Comment{jabref-meta: groups-search-syntax-version:6.0-alpha_1}
> ```
> 
> is not preserved by the pseudonymization

@f0restron07 could you fix that? I know this PR has been hanging around for long - so feel free to just address @InAnYan's comments here and fix this in a follow-up PR.

Edit - I am fine if this search groups is left as-is. As per discussion with @InAnYan on call, this needs to be thought about a bit more if it carries any semantic meaning (and thus if it also needs to be pseudonymized).

**Comment by f0restron07** [comment]
@InAnYan i have extracted the group_prefix const and also reformatted Pseudonymizationtest setup/action/check blocks with blank lines and @subhramit i have left the search group syntax fix as is pending and lmk if any changes needed further.

**Review by InAnYan — APPROVED** [review]

**Review by subhramit — APPROVED** [review]
